Add blogpost (#780)
reasonml/reason-react: website/blog/2023-09-13-reason-react-ppx.md

---

> title: reason-react-ppx is on opam!

Reviewer: ```suggestion
title: `reason-react-ppx` is now on opam!
```

---

> ReasonReact 0.12 provides the JSX transformation from `reason-react-ppx` (previously named `reactjs-jsx-ppx`) with a bunch of improvements.
> 
> The biggest changes are the new JSX transformation from React 17 and a great improvement on locations for props. A few bug fixes and some small breaking changes are also included (check below for more details).

Reviewer: ```suggestion
Today, we're releasing ReasonReact 0.12! The most important changes are renaming the JSX transform PPX from `reactjs-jsx-ppx` to `reason-react-ppx`, along with many other improvements:

- `reason-react-ppx` now emits the React 17 [JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html).
- prop locations have been greatly improved, resulting in much better editor integration.
- check the full changes below for more details around other fixes and small breaking changes.
```

### Locations from domProps in action (Thanks @jchavarri)
![download](https://github.com/reasonml/reason-react/assets/3763599/db505599-8fee-4889-80a3-d2056ece291c)

### new JSX transform (Thanks @anmonteiro)
<img width="1508" alt="Screenshot 2023-09-13 at 16 11 17" src="https://github.com/reasonml/reason-react/assets/3763599/b7fec116-131a-48cd-8be1-1496544131d1">

## What's changing?

> Track `reacjs-jsx-ppx` in the same repository as `reason-react`, and rename it to `reason-react-ppx`, a more suitable name with a few improvements:

Reviewer: would be nice to link the issues  / PRs that fixed these

Author: The full changelog is down, do you want the repetition?

- Use the [new JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) from React 17
- Fix code-generation locations for JSX and props

> - Fix to allow memoCustomCompareProps

Reviewer: ```suggestion
- Fix to allow using `React.memoCustomCompareProps`
```

- Fix Type of optional prop not checked correctly in JSX
- The ppx will be synced with reason-react (and tested together)

## Updating to latest

> To use ReasonReact, you must be using Melange. [Get started here](https://melange.re/v1.0.0/getting-started/).

Reviewer: ```suggestion
ReasonReact works best with Melange. [Get started here](https://melange.re/v1.0.0/getting-started/).
```

Next:

- Update opam repository with `opam update`
- Install the latest versions: `opam install reason-react reason-react-ppx`

> - In your dune file, add `reason-react` to `libraries` and `reason-react-ppx` to preprocess:

Reviewer: ```suggestion
- In your dune file, add `reason-react` to `libraries` and `reason-react-ppx` to `preprocess`:
```

> (libraries reason-react)
> (preprocess (pps reason-react-ppx))

- Remove any usage of `ReactDOM.props`

> - Bump React.js version to v17/v18 to use the new JSX transformation

Reviewer: should we mention check-npm-deps here?

Author: :+1:

- check the full changes below for more details around other fixes and small breaking changes.

### Locations from domProps in action (Thanks @jchavarri)
![download](https://github.com/reasonml/reason-react/assets/3763599/db505599-8fee-4889-80a3-d2056ece291c)

### new JSX transform (Thanks @anmonteiro)
<img width="1508" alt="Screenshot 2023-09-13 at 16 11 17" src="https://github.com/reasonml/reason-react/assets/3763599/b7fec116-131a-48cd-8be1-1496544131d1">

## What's changing?

Track `reacjs-jsx-ppx` in the same repository as `reason-react`, and rename it to `reason-react-ppx`, a more suitable name with a few improvements:
- Use the [new JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) from React 17
- Fix code-generation locations for JSX and props
- Fix to allow using `React.memoCustomCompareProps`
- Fix Type of optional prop not checked correctly in JSX
- The ppx will be synced with reason-react (and tested together)

## Updating to latest

ReasonReact works best with Melange. [Get started here](https://melange.re/v1.0.0/getting-started/).

Next:

> - Update opam repository with `opam update`

Reviewer 2: ```suggestion
- Update the package definitions with `opam update`
```

> - Install the latest versions: `opam install reason-react reason-react-ppx`

Reviewer 2: Should this be

```suggestion
- Upgrade to the latest versions: `opam upgrade reason-react reason-react-ppx`
```

? If this is a flow where the user already had the `reason-react` package, I'm not sure that running `opam install` will perform the upgrade.

In case they haven't installed the packages, `opam upgrade` asks the users to install them first:

```
$ opam upgrade reason-react reason-react-ppx
reason-react-ppx is not installed. Install it? [Y/n]
```

Author: Technically

opam install reason-react-ppx
opam upgrade reason-react

Reviewer 2: Yeah, maybe the first command already upgrades `reason-react` because of the `{ = version }` constraint. I suggested single `upgrade` command because it takes care of all cases at once.

---

> title: `reason-react-ppx` is now on opam!

Author: I like it

`Support for the JSX transform under a new name 'reason-react-ppx'`

Track `reacjs-jsx-ppx` in the same repository as `reason-react`, and rename it to `reason-react-ppx`, a more suitable name with a few improvements:
- Use the [new JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) from React 17
- Fix code-generation locations for JSX and props
- Fix to allow using `React.memoCustomCompareProps`
- Fix Type of optional prop not checked correctly in JSX
- The ppx will be synced with reason-react (and tested together)

Take a look at [reasonml/reason-react/releases/tag/0.12.0](https://github.com/reasonml/reason-react/releases/tag/0.12.0) for a full listing of all the changes.

## Updating to latest

ReasonReact works best with Melange. [Get started here](https://melange.re/v1.0.0/getting-started/).

Next:

- Update opam repository with `opam update`
- Install the latest versions: `opam install reason-react reason-react-ppx`
- In your dune file, add `reason-react` to `libraries` and `reason-react-ppx` to `preprocess`:

> (libraries reason-react)
> (preprocess (pps reason-react-ppx))

- Remove any usage of `ReactDOM.props`
- Bump React.js version to v17/v18 to use the new JSX transformation

> - We added `depexts` in opam for `react` and `react-dom` to ensure you have the correct versions installed from npm. [`depexts`](https://opam.ocaml.org/packages/opam-depext/) is the mechanism by opam to ensure the correct versions of external dependencies are present, often used in OS development. @jchavarri created [the tool](https://github.com/jchavarri/opam-check-npm-deps) to ensure npm packages are present.

Author: Even without documentation/page on your plugin, it's worth mentioning here since it's a good addition

> - We added `depexts` in opam for `react` and `react-dom` to ensure you have the correct versions installed from npm. [`depexts`](https://opam.ocaml.org/packages/opam-depext/) is the mechanism by opam to ensure the correct versions of external dependencies are present, often used in OS development. @jchavarri created [the tool](https://github.com/jchavarri/opam-check-npm-deps) to ensure npm packages are present.

Reviewer 2: Maybe we can link to https://melange.re/unstable/package-management/#bindings-and-package-management directly from here.

> - We added `depexts` in opam for `react` and `react-dom` to ensure you have the correct versions installed from npm. [`depexts`](https://opam.ocaml.org/packages/opam-depext/) is the mechanism by opam to ensure the correct versions of external dependencies are present, often used in OS development. @jchavarri created [the tool](https://github.com/jchavarri/opam-check-npm-deps) to ensure npm packages are present.

Author: Done, but I don't like to point to unstable links on those blog posts

>   ```clojure

Author: Looks cuter, lolaso
